pkg/kubelet/deviceplugin/manager_test.go

```diff
@@ -17,43 +17,75 @@ limitations under the License.
 package deviceplugin
 
 import (
-	"os"
-	"path"
 	"testing"
+	"time"
 
 	"github.com/stretchr/testify/require"
 
 	pluginapi "k8s.io/kubernetes/pkg/kubelet/apis/deviceplugin/v1alpha1"
 )
 
 const (
-	msocketName = "/tmp/server.sock"
+	socketName       = "/tmp/device_plugin/server.sock"
+	pluginSocketName = "/tmp/device_plugin/device-plugin.sock"
+	testResourceName = "fake-domain/resource"
 )
 
 func TestNewManagerImpl(t *testing.T) {
-	wd, _ := os.Getwd()
-	socket := path.Join(wd, msocketName)
-
 	_, err := NewManagerImpl("", func(n string, a, u, r []*pluginapi.Device) {})
 	require.Error(t, err)
 
-	_, err = NewManagerImpl(socket, func(n string, a, u, r []*pluginapi.Device) {})
+	_, err = NewManagerImpl(socketName, func(n string, a, u, r []*pluginapi.Device) {})
 	require.NoError(t, err)
 }
 
 func TestNewManagerImplStart(t *testing.T) {
-	wd, _ := os.Getwd()
-	socket := path.Join(wd, msocketName)
+	setup(t, []*pluginapi.Device{}, func(n string, a, u, r []*pluginapi.Device) {})
+}
 
-	_, err := NewManagerImpl(socket, func(n string, a, u, r []*pluginapi.Device) {})
+func TestDevicePluginReRegistration(t *testing.T) {
```

Comment:
Can you add a comment specifying what is the expected behavior and what condition(s) in re-registration logic this test is guarding against? It's not clear from the comments in the test.

Reply:
done.